src/components/button/button.tsx

```diff
@@ -59,30 +123,195 @@ export class Button {
     this.fwBlur.emit();
   }
 
-  private handleClick() {
+  private handleClick(ev) {
+    /**
+     * Handle dropdown option click
+     */
+    if (ev.path[0].nodeName === 'LI') {
+      const option = this.options.find(opt => opt.label === ev.path[0].innerText);
+      return this.handleOptionClick(option);
+    }
+
     if (this.modalTriggerId !== '') {
       const modal: any = document.getElementById(this.modalTriggerId);
       modal.visible = true;
     }
+
+    /**
+     * Do not emit event if the button is
+     * a non-split dropdown button
+     */
+
+    if (this.isValidDropdown && !this.split) {
+      return this.handleDropdownToggle();
+    }
+
     this.fwClick.emit();
   }
 
+  private hasOptions() {
```

Comment:
Can we use named slots and combine it with fw-option maybe? we can discuss on this.

Reply:
Is there an `fw-option`. Do you mean `fw-select-option`?